### DISCUSSION TITLE
melonJS 9.0.0

### DISCUSSION THREAD

**Discussion by obiot** [melonjs/melonJS#1052] [Announcements]
As we are ironing out the latest bugs and preparing to publish this release, I just wanted to share a few things on what has been going out on melonJS. 

melonJS 9.0 definitely marks the transition to the "next-gen" version of melonJS. In this version a lots of the code have been rewritten, either to follow ES6 in terms of semantic and class definition, but as well as we rewrote some fundamental components like the bounds management, that was causing issues for a while.

The build process has been also completely revamped (using Rollup and Babel), allowing to automatically build and transpile a ES5 version of melonJS out of the new code base, that is still fully backward compatible with the previous version, but while also offering a new ES6 version.

melonJS (9.x) will therefore offer the following build :

| build               | description |
| ------------------- | ----------- |
| `melonjs.js`        | the legacy ES5 UMD Bundle (directly transpiled from the ES6 version) including full API backward compatibility with previous versions |
| `melonjs.min.js`    | a minified version of the ES5 UMD bundle version |
| `melonjs-module.js` | a ES6 Module Bundle, for those living on the edge, and with no backward compatibility (note: this will break your game when upgrading!) |

Then starting from the next 10.0, all the remaining ES5 parts and things like Jay inheritance mechanism will be completely removed from melonJS, to allow offering a true modern, ES6+ and pure module based implementation.

Therefore :
* the melonJS 9.x branch will be renamed as the "legacy" branch, for those that still needs to use the previous ES5 UMD bundle, with all the backward API, and will keep offering bug fixing, but no new major features.
* melonJS 10.0 and higher will only provide a pure ES6 implementation and module based.  Also a new project boilerplate will be created to allow still building (Transpiling to) ES5 version of a game (using web pack most probably)

and that's it ! let us know what you think about it :)

Thanks !
Oliver.

**Comment by parasyte** [comment]
Very nice! I'm really happy to hear that ES6 classes are being taken seriously. And that the build process has been revamped. And well, everything really! You have done a lot of work maintaining melonJS, and I look forward to seeing what's next.

**Reply by obiot** [reply]
Thanks, means a lot coming from you :)
